chain/indexing.go

```diff
@@ -0,0 +1,45 @@
+package chain
+
+import (
+	"context"
+
+	"github.com/filecoin-project/lotus/chain/events"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+func NewIndexingTipSetObserver(obs TipSetObserver, cache *TipSetCache) *IndexingTipSetObserver {
```

Comment:
As you mentioned before, we could simply revert to using ChainNotify and then this type is redundant with Watcher. For now let's keep it and work on refactoring to a more optimized lens interface that could work by calling ChainNotiify or could hook into events directly.